sql/core/src/test/resources/sql-tests/inputs/subexp-elimination.sql

```diff
@@ -0,0 +1,37 @@
+-- Test for subexpression elimination.
+
+--SET spark.sql.optimizer.enableJsonExpressionOptimization=false
+
+--CONFIG_DIM1 spark.sql.codegen.wholeStage=true
+--CONFIG_DIM1 spark.sql.codegen.wholeStage=false
+
+--CONFIG_DIM2 spark.sql.codegen.factoryMode=CODEGEN_ONLY
+--CONFIG_DIM2 spark.sql.codegen.factoryMode=NO_CODEGEN
+
+--CONFIG_DIM3 SUBEXPRESSION_ELIMINATION_ENABLED=true
+--CONFIG_DIM3 SUBEXPRESSION_ELIMINATION_ENABLED=false
```

Comment:
Wait, shouldn't it be the actual config names?

Reply:
Oops, let me fix it.

Reply:
Fix in #30384.